### DISCUSSION TITLE
How to process a backend GET request, and properly show a frontend page?

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#68110] [Help] [answered]
### Summary

I am trying to do an unsubscribe page that would be handle in a single click from the user.

When the user makes a click to unsubscribe from his email link, he loads a page and sends a communication to the server, and the server handles a GET request with parameters to do some backend operations.   After the backend is done, I need to show some simple html on the  user screen.

Something like:

```
return new Response(
    (
      <div>
        <h1>Thank you.</h1>
        <p> Your subscription has been updated.  </p>
      </div>
    )
  );
```

I suppose I could also do some type of forwarding to `/this-is-another-page`  type of thing, but it would be kind of bouncing a lot?

**Comment by Juneezee** [comment, marked as answer]
There are a few ways that you can implement this.

1. `redirect` in Route Handler. This is what you have described in your description.
2. Handle unsubscription in a Server Component `page.tsx`

   ```tsx
   export default async function Page({
     searchParams: { email },
   }: {
     searchParams: { email?: string | undefined };
   }): Promise<React.JSX.Element> {
     const success = email !== undefined && (await unsubscribeEmail(email));

     return (
       <div>{success ? <p>You have been unsubscribed.</p> : <p>Error</p>}</div>
     );
   }
   ```

   Here the `unsubscribeEmail` is a Server Action and `page.tsx` is a Server Component. You can read more about this at https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions-and-mutations#server-components

**Reply by Elindo586** [reply]
Hello!  

Thank you much for your reply.   

I was looking at the re-direct, but I didn't know how to pass the  parameter value to the the front end.. so I ended up doing some like this:

```
const htmlResponse = `
      <html>
        <body style = "color: blue">
          <div>
            <div> <h1> Hello, </h1> </div>
            <div> Your subscription for email: ${email} has been updated. </div>
          </div>
        </body>
      </html>
    `;


  return new Response(htmlResponse, {
    headers: { "Content-Type": "text/html" },
  });
```

^^^   I am just testing this, but I can add html and CSS for just a simple screen.

I will be looking at your solution as well.

Thank you much!
